### PR TITLE
fix(tooltip): fix infinite loop in self-managed tooltips

### PR DESCRIPTION
**Pull request by Rocss**
## Description

Reordering some `if` statements to make sure a trigger is always returned for self-managed tooltips.

## Related issue(s)

- https://github.com/adobe/spectrum-web-components/issues/4268

## Motivation and context

Fixes a bug which reproduces only when NOT in DEBUG mode. See details in the issue above.

Not sure how to add a unit test for this, since `triggerElement` is private and I can not check it from outside.

## How has this been tested?

-   [x] window.__swc.DEBUG === false
    1. Modified the `self-managed` story and replaced `sp-action-button` with a `div`
    2. No infinite loop - trigger ends up being the `document`
    3. No warning is thrown in the console
-   [x] window.__swc.DEBUG === true
    1. Modified the `self-managed` story and replaced `sp-action-button` with a `div`
    2. No infinite loop - trigger ends up being the `document`
    4. Warning is thrown in the console

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Review by Westbrook — COMMENTED** [review]
Do we have tests that prove that _if_ it finds document it _doesn't_ bind listeners? A Tooltip in this situation should not work in any way, it should only dispatch the Dev Mode warning.

**Comment by Rocss** [comment]
@Westbrook As it is right now it _does_ bind event listeners on the document.
May I suggest that `get triggerElement()` would return `null` if no focusable element was found in the parent chain, instead of `document`?
Looking at the `Overlay` class, it seems like it is allowed to be `null`:
```
Overlay.triggerElement: HTMLElement | VirtualTrigger | null
```

**Review by Westbrook — COMMENTED** [review]
Do we have tests that prove that _if_ it finds document it _doesn't_ bind listeners? A Tooltip in this situation should not work in any way, it should only dispatch the Dev Mode warning.

If not, we can land this on the presumption it works, as it seems to not be working for you otherwise.

File: packages/tooltip/test/tooltip.test.ts [hunk 1]
```diff
@@ -214,6 +214,19 @@ describe('Tooltip', () => {
...
+    it('self managed does not attach event listeners if no trigger was found', async () => {
+        const documentEventsSpy = spy(document, 'addEventListener');
+        const el = await fixture<HTMLDivElement>(
+            html`
+                <sp-tooltip self-managed>Help text.</sp-tooltip>
+            `
+        );
+
+        await elementUpdated(el);
+        expect(documentEventsSpy.callCount).to.equal(0);
```
Comment (Westbrook) [inline packages/tooltip/test/tooltip.test.ts:227]: Can we confirm it doesn't attach to anything else, too?
```suggestion
        expect(documentEventsSpy.callCount).to.equal(0);
        expect(el.overlayElement?.triggerElement).to.be.null;
```

**Review by Westbrook — COMMENTED** [review]
For Lighthouse/a preview URL, your branch name is too long. 🙈 We won't count it against you 😉 

This is a good test, with that one extra expectation and a rebase to `main` we should be good to go here. Thanks for finding tis issue!

**Review by Westbrook — APPROVED** [review]
:shipit: Thank you, thank you!

**Pull request merged**